### ISSUE TITLE
Maximum Limit of 1.4 MB

### ISSUE THREAD

**Issue by just-kirb** [Auroriax/PuzzleScriptPlus#126]
**Describe the bug**
A game cannot store over 1.4MB of data in a link as it starts deleting the ending lines of code.

**Reproduction Steps**
1. Make a game with lots of rules to check and exceed 1.4MB of data.
2. Share the game with the link.
3. Open it.

**What did you expect would happen?**
The data being intact after sharing the link to others.

**OS & Browser version, desktop/mobile**
Edition: Windows 10 Home Single Language
OS Version: 22H2
Firefox Version 129.0.2 (64-bit)

**Screenshots, example projects, other relevant issues, etc.**
![image](https://github.com/user-attachments/assets/94fb4d9c-0cf0-4f3b-9272-3407fc471171)
information before sharing the link
![image](https://github.com/user-attachments/assets/7161475d-212b-4a89-a2a9-b2838c9fc238)
information removed when sharing the link

**Comment by david-pfx** [comment]
Data? Do you means lines of script code? That would be maybe 50 KLOC. Do games that big actually work?

**Comment by just-kirb** [comment]
I mean the games up until that point work fine when programming, however the issue probably comes from GitHub's limit on storing data in links (i verified saving the game doesn't cause an override, but sharing it though a link does), here is a video demonstrating the problem more clearly:

https://github.com/user-attachments/assets/734fde5e-80c5-4b31-97b1-b650ff957b8c

**Comment by just-kirb** [comment]
UPDATE: i feel like the puzzlescript machine for github gists only scraped a truncated portion of the script, and not the actual full version, causing a cutoff, but the full file is still intact

**Comment by david-pfx** [comment]
I’m happy to look at it, but I do need a repro. The Gist maybe.

 

Regards

David M Bennett FACS

  _____  

Polyomino Games – Programming Languages and Players for Games and Puzzles -- http://www.polyomino.com <http://www.polyomino.com/>

**Comment by just-kirb** [comment]
https://gist.github.com/just-kirb/dd9d7807216b610c716a5ed4e7c5340c

**Comment by david-pfx** [comment]
Yes, it's real. It seems to affect every fork. It truncates in line 22977.

Is it a Github problem/limit? In any case, I don't this is fixable.

**Comment by just-kirb** [comment]
wait the github program that runs puzzlescript plus doesn't have a way to
scrape gists fully if it's truncated?
